test/engine/BindTest.cpp

```diff
@@ -0,0 +1,116 @@
+//   Copyright 2024, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#include <gtest/gtest.h>
+
+#include "../util/IdTableHelpers.h"
+#include "../util/IndexTestHelpers.h"
+#include "./ValuesForTesting.h"
+#include "engine/Bind.h"
+#include "engine/sparqlExpressions/LiteralExpression.h"
+
+using namespace sparqlExpression;
+using Vars = std::vector<std::optional<Variable>>;
+
+namespace {
+Bind makeBindForIdTable(QueryExecutionContext* qec, IdTable idTable) {
+  auto valuesTree = ad_utility::makeExecutionTree<ValuesForTesting>(
+      qec, std::move(idTable), Vars{Variable{"?a"}});
+  return {
+      qec,
+      std::move(valuesTree),
+      {SparqlExpressionPimpl{
+           std::make_unique<VariableExpression>(Variable{"?a"}), "?a as ?b"},
+       Variable{"?b"}}};
+}
+
+IdTable getSingleIdTable(cppcoro::generator<IdTable>& generator) {
+  std::optional<IdTable> result = std::nullopt;
+  for (IdTable& idTable : generator) {
+    if (result.has_value()) {
+      ADD_FAILURE() << "More than one IdTable was generated";
+      break;
+    }
+    result = std::move(idTable);
+  }
+  if (!result.has_value()) {
+    throw std::runtime_error{"No IdTable was generated"};
+  }
+  return std::move(result).value();
+}
```

Comment:
I think this can become much simpler if you use `begin()` and `end()` on the generator explicitly.
Also we now have this helper function in `Generator.h`, you could just reuse it. In that case you wouldn't get the `ADD_FAILURE()` but an exception, but that is your decision, your second failure case also throws here.